cmd/adapter/adapter.go

```diff
@@ -334,12 +335,27 @@ func main() {
 		klog.Fatalf("unable to install resource metrics API: %v", err)
 	}
 
+	config, err := cmd.Config()
+	if err != nil {
+		klog.Fatalf("unable to get configuration of custom metrics apiserver: %v", err)
+	}
+	setupMetricsServer(config)
+
 	// run the server
 	if err := cmd.Run(stopCh); err != nil {
 		klog.Fatalf("unable to run custom metrics adapter: %v", err)
 	}
 }
 
+// Disables the metrics endpoint in the api server and brings up a new metrics endpoint
+func setupMetricsServer(config *customexternalmetrics.Config) {
+	config.GenericConfig.EnableMetrics = false
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", legacyregistry.Handler())
+
+	go http.ListenAndServe(":9090", mux)
```

Comment:
The server address and port should be configurable if we want to be consistent with what was previously available. Also, since the generic apiserver was serving the metrics over HTTPS, we should do the same with the new server.

Reply:
Done